Upstream fix of https://github.com/openssl/openssl/pull/17595

Any propquery passed via the -propquery option to the ciphers command was
being ignored.

--- openssl-3.0.1.orig/apps/ciphers.c	2022-01-26 15:10:03.043518289 +0000
+++ openssl-3.0.1/apps/ciphers.c	2022-01-26 15:10:55.134506382 +0000
@@ -187,7 +187,7 @@
         goto end;
     }
 
-    ctx = SSL_CTX_new(meth);
+    ctx = SSL_CTX_new_ex(app_get0_libctx(), app_get0_propq(), meth);
     if (ctx == NULL)
         goto err;
     if (SSL_CTX_set_min_proto_version(ctx, min_version) == 0)
